src/pages/plans/Plans.tsx

```diff
@@ -1,96 +1,241 @@
-      toast.error("Ingresa un monto válido.");
-      return;
+    return planList.filter((plan) =>
+      plan.name.toLowerCase().includes(filter.toLowerCase())
+    );
+  }, [planList, filter]);
+
+  const handleClearFilters = () => {
+    setFilter("");
+  };
+  const handleMutation = async (
+    action: () => Promise<any>,
+    onSuccess: () => void,
+    successMessage: string,
+    errorMessage: string
+  ) => {
+    try {
+      await action();
+      toast.success(successMessage);
+      onSuccess();
+    } catch (error: any) {
+      toast.error(errorMessage);
     }
+  };
+  //CREAR PLAN
+  const handleCreateModal = async () => {
+    if (!formRef.current) return;
+
+    const data = await formRef.current.submit();
+    if (!data) return;
+
+    await handleMutation(
+      () => createPlan(data).unwrap(),
+      () => setShowCreateModal(false),
+      "Plan creado correctamente",
+      "Error al crear el plan"
+    );
+  };
+  //EDITAR PLAN
+  const handleEditModal = async () => {
+    if (!formRef.current) return;
+
+    const data = await formRef.current.submit();
+    if (!data) return;
 
-    updatePrice({
-      priceId: selectedPriceId,
-      amount: Number(newAmount),
-    })
-      .unwrap()
-      .then(() => {
-        toast.success("Precio actualizado con éxito");
-        setNewAmount(0);
-      });
+    await handleMutation(
+      () => editPlan(data).unwrap(),
+      () => setShowEditModal(false),
+      "Plan editado correctamente",
+      "Error al editar el plan"
+    );
   };
+  //ELIMINAR PLAN
+  const handleDeleteModal = async () => {
+    if (!selectedPlanId) return;
 
+    await handleMutation(
+      () => deletePlan(selectedPlanId).unwrap(),
+      () => {
+        setShowDeleteModal(false);
+        setSelectedPlanId(null);
+      },
+      "Plan eliminado correctamente",
+      "Error al eliminar el plan"
+    );
+  };
+
+  const columns: Column<PlanResponse>[] = [
+    {
+      header: <SortableButton text="Nombre del plan" />,
+      accessor: "name",
+    },
+    {
+      header: <SortableButton text="Cantidad de días asignados" />,
+      accessor: "numberOfDays",
+      render: (plan) => <span>{plan.numberOfDays}</span>,
+    },
+    {
+      header: <SortableButton text="Precio actual" />,
+      accessor: "price",
+      render: (plan) => <span>{formatCurrency(plan.price)}</span>,
+    },
+    {
+      header: "Acciones",
+      render: (plan) => (
+        <DropdownMenu
+          icon={<img src={DotsIcon} alt="Opciones" width={30} height={30} />}
+          size="small"
+          options={[
+            {
+              label: "Editar plan",
+              onClick: () => {
+                setShowEditModal(true);
+                setSelectedPlan(plan);
+              },
+            },
+            {
+              label: "Eliminar",
+              onClick: () => {
+                setShowDeleteModal(true);
+                setSelectedPlanId(plan.id);
```

Comment:
Y aca tendrias que tener


```suggestion
                setSelectedPlan(plan);
```

De esta manera te evitar tener un useState mas al pedo